public/test/test-utils.tsx

```diff
@@ -0,0 +1,111 @@
+import { ToolkitStore } from '@reduxjs/toolkit/dist/configureStore';
+import { render, RenderOptions } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { KBarProvider } from 'kbar';
+import React from 'react';
+import { Provider } from 'react-redux';
+import { MemoryRouter } from 'react-router-dom';
+import { PreloadedState } from 'redux';
+import { getGrafanaContextMock } from 'test/mocks/getGrafanaContextMock';
+
+import { config } from '@grafana/runtime';
+import { ErrorBoundaryAlert } from '@grafana/ui';
+import { GrafanaContext, GrafanaContextType } from 'app/core/context/GrafanaContext';
+import { ModalsContextProvider } from 'app/core/context/ModalsContextProvider';
+import { ThemeProvider } from 'app/core/utils/ConfigProvider';
+import { configureStore } from 'app/store/configureStore';
+import { StoreState } from 'app/types/store';
+
+interface ExtendedRenderOptions extends RenderOptions {
+  /**
+   * Partial state to use for preloading store when rendering tests
+   */
+  preloadedState?: PreloadedState<StoreState>;
+  /**
+   * Optional
+   */
+  store?: ToolkitStore;
+  renderWithRouter?: boolean;
+}
+
+/**
+ * Wraps children in empty fragment - used to conditionally render in a router or not,
+ * as needed by tests
+ */
+const FragmentWrapper = ({ children }: { children: React.ReactNode }) => {
+  return children;
+};
+
+/**
+ * Get a wrapper component that implements all of the providers that components
+ * within the app will need
+ */
+const getWrapper = ({
+  store,
+  renderWithRouter,
+  grafanaContext,
+}: {
+  store?: ToolkitStore;
+  /**
+   * Should the wrapper be generated with a wrapping Router component?
+   * Useful if you're testing something that needs more nuanced routing behaviour
+   * and you want full control over it instead
+   */
+  renderWithRouter?: boolean;
+  grafanaContext?: GrafanaContextType;
+}): React.FC<{ children: React.ReactNode }> => {
```

Comment:
```suggestion
}): React.FC<PropsWithChildren> => {
```
Can we use `PropsWithChildren` here?

Reply:
:+1: Will change to `PropsWithChildren` throughout, thanks